internal/civisibility/integrations/gotesting/reflections.go

```diff
@@ -36,7 +97,108 @@ func getInternalTestArray(m *testing.M) *[]testing.InternalTest {
 	return nil
 }
 
+// getTestPrivateFields is a method to retrieve all required privates field from
+// testing.T, returning a commonPrivateFields instance
+func getTestPrivateFields(t *testing.T) *commonPrivateFields {
+	testFields := &commonPrivateFields{}
+
+	// testing.common
+	if ptr, err := getFieldPointerFrom(t, "mu"); err == nil {
+		testFields.mu = (*sync.RWMutex)(ptr)
+	}
+	if ptr, err := getFieldPointerFrom(t, "level"); err == nil {
+		testFields.level = (*int)(ptr)
+	}
+	if ptr, err := getFieldPointerFrom(t, "name"); err == nil {
+		testFields.name = (*string)(ptr)
+	}
+	if ptr, err := getFieldPointerFrom(t, "failed"); err == nil {
+		testFields.failed = (*bool)(ptr)
+	}
+	if ptr, err := getFieldPointerFrom(t, "skipped"); err == nil {
+		testFields.skipped = (*bool)(ptr)
+	}
+	if ptr, err := getFieldPointerFrom(t, "parent"); err == nil {
+		testFields.parent = (*unsafe.Pointer)(ptr)
+	}
+
+	return testFields
+}
+
+// getTestParentPrivateFields is a method to retrieve all required parent privates field from
+// testing.T.parent, returning a commonPrivateFields instance
+func getTestParentPrivateFields(t *testing.T) *commonPrivateFields {
+	indirectValue := reflect.Indirect(reflect.ValueOf(t))
+	member := indirectValue.FieldByName("parent")
+	if member.IsValid() {
+		value := member.Elem()
+		testFields := &commonPrivateFields{}
+
+		// testing.common
+		if ptr, err := getFieldPointerFromValue(value, "mu"); err == nil {
+			testFields.mu = (*sync.RWMutex)(ptr)
+		}
+		if ptr, err := getFieldPointerFromValue(value, "level"); err == nil {
+			testFields.level = (*int)(ptr)
+		}
+		if ptr, err := getFieldPointerFromValue(value, "name"); err == nil {
+			testFields.name = (*string)(ptr)
+		}
+		if ptr, err := getFieldPointerFromValue(value, "failed"); err == nil {
+			testFields.failed = (*bool)(ptr)
+		}
+		if ptr, err := getFieldPointerFromValue(value, "skipped"); err == nil {
+			testFields.skipped = (*bool)(ptr)
+		}
+
+		return testFields
+	}
+	return nil
+}
+
+// copyTestWithoutParent tries to copy all private fields except the t.parent from a *testing.T to another
+func copyTestWithoutParent(source *testing.T, target *testing.T) {
```

Comment:
I'm not sure about this, but can't we just do `*target = *source` and then overwrite just the `parent` field?

Reply:
no, we need a `clean` testing.T for each test execution. If we reuse the same `t` statuses may be contaminated as the subtest list, etc...

Reply:
But this function copies the subtest list as well, so isn't it subject to the same issue?

Also, I mean doing `*target = *source`, not `target = source` (i.e., copying the contents, not the pointers). But now I'm worried about the nested maps and slices as well (regardless of how the fields are copied) :grimacing: Sorry if I'm missing something here.

Reply:
yeah I tried to make a normal content copy, but `go vet` complains because `t` has a mutex, so it was copying the mutex value. There's no way to opt-out the warning and the CI complains about that. So I opted to do a manual copy.

Nested maps and slices can be copied, because the `source` doesn't have any execution yet when we do the copy, so is copying empty maps and slices... We can skip those from being copied as well we no change of behaviour.

My reference to a `clean` testing.T was about not reusing the same reference to T from 1 execution to another, but always creating a new copy from the `source` for each new execution.

Reply:
Thanks for the explanation!